server/src/main/java/org/apache/druid/segment/join/InlineJoinableFactory.java

```diff
@@ -39,8 +38,7 @@
   {
     if (condition.canHashJoin() && dataSource instanceof InlineDataSource) {
       final InlineDataSource inlineDataSource = (InlineDataSource) dataSource;
-      final List<String> rightKeyColumns =
-          condition.getEquiConditions().stream().map(Equality::getRightColumn).distinct().collect(Collectors.toList());
+      final List<String> rightKeyColumns = condition.getRightKeyColumns();
```

Comment:
You could probably do a similar simplification in LookupJoinMatcher (there's a part that checks that all the equikeys are the key column).

Reply:
It's not immediately obvious to me how other implementations would use what `LookupJoinMatcher` is doing with the equiConditions. If it's ok with you, I'd like to do that refactoring at a later time

Reply:
Ah, I just meant replacing this:

```java
    } else if (!condition.getEquiConditions()
                         .stream()
                         .allMatch(eq -> eq.getRightColumn().equals(LookupColumnSelectorFactory.KEY_COLUMN))) {
```

With this:

```java
    } else if (!condition.getRightEquiConditionKeys().equals(Collections.singletonSet(LookupColumnSelectorFactory.KEY_COLUMN)) {
```